test/e2e/e2e.go

```diff
@@ -112,19 +110,19 @@ func RunE2ETests(t *testing.T) {
 	}
 
 	// Run tests through the Ginkgo runner with output to console + JUnit for Jenkins
-	var r []ginkgo.Reporter
 	if framework.TestContext.ReportDir != "" {
 		// TODO: we should probably only be trying to create this directory once
 		// rather than once-per-Ginkgo-node.
+		// NOTE: junit report can be simply created by executing your tests with the new --junit-report flags instead.
+		// framework.TestContext.ReportDir and framework.TestContext.ReportPrefix are deprecated and will be removed in 1.28,
+		// this directory is created in order to not break the saving of image.json and Pods' event log.
```

Comment:
let me see if I understand this change, so, the flags exist but they don't do anything?

Reply:
they won't be used for junit report, but they are still widely used for other purpose, it will hurt if I just remove this flag, my suggestion is we can set some warning message for those two flags and remove them +3 releases later.
https://grep.app/search?q=ReportDir&filter[repo][0]=kubernetes/kubernetes

Reply:
See also my comments about this in https://github.com/kubernetes/kubernetes/pull/109111#pullrequestreview-963208732

If we keep the logic as it is now, we need to:
- extend the changenote to mention how to generate JUnit output now
- plan and explain what will replace these command line flags

Reply:
How hard is it to enable the JUnit reporter when a report dir is specified? That would be the simplest solution for our users (= no disruption at all).

Reply:
thanks @pohly , will update the releasenote in my next day.

Reply:
So the flag won't be deprecated, it will just behave differently?

Have you checked whether the JUnit report can be enabled via code when a report dir is set?

Reply:
I've looked into this and came to the (same?) conclusion that this isn't going to be easy. Let's document the new command line flag and keep --report-dir as is with reduced functionality (used for some of the other output files, but not for JUnit).

Reply:
happy to help think about it.  what's the upshot?  can y'all summarize the problem?

Reply:
e2e.test has a `-report-dir` flag. If such a directory is specified, JUnit files were created there. We need to figure out whether it is possible to do the same with ginkgo v2 or whether we have to tell customers that they must update their invocation of the e2e.test suite.

We already have to ask users to update their invocation, so adding one more change in those instructions is not the end of the world. It just would be simpler if we didn't have to do this.

Reply:
got it.  can you point me at the invocation instructions?  that will probably answer some questions that come to mind.

Reply:
Parameter definition:
https://github.com/kubernetes/kubernetes/blob/784cf764690917e6fb229532722ea0d857cfcde4/test/e2e/framework/test_context.go#L304

Usage:
https://github.com/kubernetes/kubernetes/blob/784cf764690917e6fb229532722ea0d857cfcde4/test/e2e/e2e.go#L108-L118

Reply:
> So the flag won't be deprecated, it will just behave differently?

Have already set the flags as deprecated (https://github.com/kubernetes/kubernetes/pull/109111/commits/c0eb85127e55018b38e04b0c3bbc0a1fbf129434), is that not enough?